firewood/src/merkle/node.rs

```diff
@@ -375,8 +384,27 @@ impl Storable for Node {
             None
         };
 
+        let mut start_index = TRIE_HASH_LEN + 1;
+        let end_index = start_index + size_of::<u64>();
+        #[allow(clippy::indexing_slicing)]
+        let encoded_len = u64::from_le_bytes(
+            meta_raw.as_deref()[start_index..end_index]
+                .try_into()
+                .expect("invalid slice"),
```

Comment:
A `Cursor` here with `read_exaxt` would make it so you wouldn't need to index. You would just create a buffer of the proper size and read into that. A good example can be seen in `branch.rs`.